src/reason-parser/reason_parser.mly

```diff
@@ -1355,18 +1355,14 @@ conflicts.
 
 (* PREFIXOP and BANG precedence *)
 %nonassoc below_DOT_AND_SHARP           (* practically same as below_SHARP but we convey purpose *)
-%left    MINUSGREATER
 %nonassoc SHARP                         (* simple_expr/toplevel_directive *)
 %nonassoc below_DOT
-%nonassoc DOT POSTFIXOP
 
 (* We need SHARPEQUAL to have lower precedence than `[` to make e.g.
    this work: `foo #= bar[0]`. Otherwise it would turn into `(foo#=bar)[0]` *)
 %left     SHARPEQUAL
-
-%nonassoc LBRACKET
-(* SHARPOP has higher precedence than `[`, see e.g. issue #1507. *)
-%left     SHARPOP
+%left     SHARPOP MINUSGREATER LBRACKET DOT
```

Comment:
Is it possible to add a comment that explains the road that lead us to making all the `%left` assoc?